litellm/proxy/db/db_transaction_queue/spend_log_cleanup.py

```diff
@@ -56,35 +64,63 @@ async def cleanup_old_spend_logs(self, prisma_client: PrismaClient) -> None:
                 verbose_proxy_logger.error("Retention seconds is None, cannot proceed with cleanup")
                 return
 
-            cutoff_date = datetime.now(UTC) - timedelta(seconds=float(self.retention_seconds))
-            verbose_proxy_logger.info(f"Deleting logs older than {cutoff_date.isoformat()}")
-
-            total_deleted = 0
-            run_count = 0
-            while True:
-                if run_count > 100:
-                    verbose_proxy_logger.info("Max logs deleted - 1,00,000, rest of the logs will be deleted in next run")
-                    break
-                # Step 1: Find logs to delete
-                logs_to_delete = await prisma_client.db.litellm_spendlogs.find_many(
-                    where={"startTime": {"lt": cutoff_date}},
-                    take=self.batch_size,
-                )
-                verbose_proxy_logger.info(f"🗑️ Found {len(logs_to_delete)} logs in this batch")
-
-                if not logs_to_delete:
-                    verbose_proxy_logger.info(f"No more logs to delete. Total deleted: {total_deleted}")
-                    break
-
-                request_ids = [log.request_id for log in logs_to_delete]
-
-                # Step 2: Delete them in one go
-                await prisma_client.db.litellm_spendlogs.delete_many(
-                    where={"request_id": {"in": request_ids}}
-                )
-
-                total_deleted += len(logs_to_delete)
-                verbose_proxy_logger.info(f"Deleted {len(logs_to_delete)} logs in this batch")
-                run_count += 1
+            # Check if pod_lock_manager and redis_cache exist
+            if not self.pod_lock_manager or not self.pod_lock_manager.redis_cache:
+                verbose_proxy_logger.info("Pod lock manager or redis cache not initialized, skipping cleanup")
+                return
+
+            # Try to acquire the distributed lock
+            lock_acquired = await self.pod_lock_manager.acquire_lock(cronjob_id=SPEND_LOG_CLEANUP_JOB_NAME)
+            verbose_proxy_logger.info(f"Lock acquisition attempt: {'successful' if lock_acquired else 'failed'}")
+            
+            if not lock_acquired:
+                verbose_proxy_logger.info("Another pod is already running cleanup")
+                return
+
+            try:
+                cutoff_date = datetime.now(timezone.utc) - timedelta(seconds=float(self.retention_seconds))
+                verbose_proxy_logger.info(f"Deleting logs older than {cutoff_date.isoformat()}")
+
+                total_deleted = 0
+                run_count = 0
+                while True:
+                    if run_count > 100:
```

Comment:
nit: you should probably have a smaller method that's doing the deletion for you 

and then have the lock acquire/lock release logic around your helper